src/Metadata/AssemblyMetadata.Methods.cs

```diff
@@ -8,6 +8,18 @@ namespace Lokad.ILPack.Metadata
 {
     internal partial class AssemblyMetadata
     {
+        public EntityHandle GetMethodHandleConstructedType(MethodInfo method) {
```

Comment:
Idem, you need to justify why you are introducing an alternative to `GetMethodHandle`. It strongly feels that `GetMethodHandle` should be  refactored instead.